lib/ipam/ipam_test.go

```diff
@@ -103,364 +77,323 @@ type testArgsClaimAff struct {
 	expError                    error
 }
 
-var _ = testutils.E2eDatastoreDescribe("IPAM tests", testutils.DatastoreEtcdV2, func(config api.CalicoAPIConfig) {
+var _ = testutils.E2eDatastoreDescribe("IPAM tests", testutils.DatastoreEtcdV3, func(config apiconfig.CalicoAPIConfig) {
+	// Create a new backend client and an IPAM Client using the IP Pools Accessor.
+	// Tests that need to ensure a clean datastore should invokke Clean() on the datastore at the start of the
+	// tests.
+	bc, err := backend.NewClient(config)
+	if err != nil {
+		panic(err)
+	}
+	ic := NewIPAM(bc, ipPools)
 
 	// We're assigning one IP which should be from the only ipPool created at the time, second one
 	// should be from the same /26 block since they're both from the same host, then delete
 	// the ipPool and create a new ipPool, and AutoAssign 1 more IP for the same host - expect the
 	// assigned IP to be from the new ipPool that was created, this is to make sure the assigned IP
 	// doesn't come from the old affinedBlock even after the ipPool was deleted.
 	Describe("IPAM AutoAssign from the default pool then delete the pool and assign again", func() {
-		c := testutils.CreateCleanClient(config)
-		ic := setupIPAMClient(c, true)
-
-		host := "host-A"
+		hostA := "host-A"
+		hostB := "host-B"
 		pool1 := cnet.MustParseNetwork("10.0.0.0/24")
+		pool2 := cnet.MustParseNetwork("20.0.0.0/24")
 		var block cnet.IPNet
 
-		testutils.CreateNewIPPool(*c, "10.0.0.0/24", false, false, true)
+		Context("AutoAssign a single IP without specifying a pool", func() {
 
-		// Step-1: AutoAssign 1 IP without specifying a pool - expect the assigned IP is from pool1.
-		Context("AutoAssign 1 IP without specifying a pool", func() {
-			args := client.AutoAssignArgs{
-				Num4:     1,
-				Num6:     0,
-				Hostname: host,
-			}
+			It("should auto-assign from the only available pool", func() {
+				bc.Clean()
+				deleteAllPools()
+				applyPool("10.0.0.0/24", true)
```

Comment:
how about having global strings `pool1Str := "10.0.0.0/24"` and one for pool2 so it's easy to correlate when reading the tests what's what, and it's also less error-prone

Reply:
yeah that might be nice - but this is old code and I don't want the scope of this task to keep creeping out.